File: expensereport-java/src/test/java/com/nelkinda/training/Interceptor.java
package com.nelkinda.training;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

public class Interceptor {
    public static String interceptStdout(final Runnable code) {
        final PrintStream originalStdout = System.out;
        final ByteArrayOutputStream interceptedStdout = new ByteArrayOutputStream();
        System.setOut(new PrintStream(interceptedStdout));
        try {
            code.run();
        } finally {
            System.setOut(originalStdout);
        }
        return interceptedStdout.toString();
    }
}

File: expensereport-kotlin/src/test/kotlin/com/nelkinda/training/Interceptor.kt
@file:JvmName("Interceptor")

package com.nelkinda.training

import java.io.ByteArrayOutputStream
import java.io.PrintStream

fun interceptStdout(code: () -> Unit): String {
    val originalStdout = System.out
    val interceptedStdout = ByteArrayOutputStream()
    System.setOut(PrintStream(interceptedStdout))
    try {
        code()
    } finally {
        System.setOut(originalStdout)
    }
    return interceptedStdout.toString()
}
